librecval/extract_phrases.py

```diff
@@ -140,27 +137,45 @@ def scan(self, root_directory: Path):
         """
 
         self.logger.debug("Scanning %s for sessions...", root_directory)
+
+        valid_session_directories = []
+
         for session_dir in root_directory.iterdir():
             if not session_dir.resolve().is_dir():
                 self.logger.debug("Rejecting %s; not a directory", session_dir)
                 continue
+            valid_session_directories.append(session_dir)
+
+        for session_dir in valid_session_directories:
             try:
-                yield from self.extract_session(session_dir)
-            except DuplicateSessionError:
-                self.logger.exception("Skipping %s: duplicate", session_dir)
-            except MissingMetadataError:
-                self.logger.exception("Skipping %s: Missing metadata", session_dir)
+                yield from self.extract_all_recordings_from_session(session_dir)
+            except Exception:
+                session_id = get_session_name_or_none(session_dir)
+                if session_id is None:
+                    continue
+
+                self.logger.exception("Error extracting %s", session_dir)
+                failed_dir = Path.cwd() / "failed-sessions"
```

Comment:
This will sometimes end up in random directories if you run the script from random directories. Do you want to build a path off of `Path(__file__).parent`, or maybe some `settings.BASE_DIR` constant?

Reply:
This package is supposed to be unaware that Django exists, so I'll base it on `Path(__file__).parent`!